toolkit/primitives/block-production-log/src/test.rs

```diff
@@ -36,9 +37,12 @@ sp_api::mock_impl_runtime_apis! {
 fn provides_author_based_on_runtime_api() {
 	let mock_api = TestApi { author: 102 };
 
-	let provider =
-		BlockAuthorInherentProvider::<Author>::new(&mock_api, <Block as BlockT>::Hash::default())
-			.expect("Should not fail");
+	let provider = BlockAuthorInherentProvider::<Author>::new(
+		&mock_api,
+		<Block as BlockT>::Hash::default(),
+		Slot::from(42),
+	)
+	.expect("Should not fail");
 
-	assert_eq!(provider.author.unwrap(), mock_api.author.into());
+	assert_eq!(provider.author.unwrap(), mock_api.author as Author);
```

Comment:
try to keep into if possible

Reply:
for some reason it doesn't work. `as` does the job

Reply:
probably needs some additional type annotations, try:
```suggestion
	assert_eq!(provider.author.unwrap(), Author::from(mock_api.author));
```